pandas/core/generic.py

```diff
@@ -7807,6 +7807,66 @@ def mask(self, cond, other=np.nan, inplace=False, axis=None, level=None,
         is not realigned. That is, use freq if you would like to extend the
         index when shifting and preserve the original data.
 
+        Examples
+        --------
+        Compute the difference between a column in a dataframe and
+        its shifted version
+
+        >>> data = pd.DataFrame({'mydate': [pd.to_datetime('2016-06-06'),
```

Comment:
maybe use `pd.Timestamp('2016-06-06')` or `pd.to_datetime(['2016-06-06', ...])` to convey that `to_datetime` should be used for arrays.

Reply:
Move `mydate` to `index=` and remove the `set_index` below.

Reply:
It is useful to have a named index as the label (mydate) is used later in the process (for the left join). Would you prefer if I use `index= ` and then set the label using  `data.index.name =  ` ?

Reply:
Agreed. How about 

```
index=pd.DatetimeIndex(['2016-06-08', ... '2016-06-13'], name='mydate')
```.